server/environment/environment.go

```diff
@@ -76,4 +76,5 @@ type Env interface {
 	GetUsageService() interfaces.UsageService
 	GetUsageTracker() interfaces.UsageTracker
 	GetXCodeLocator() interfaces.XcodeLocator
+	GetFileResolver() fs.FS
```

Comment:
Since we already have:
	GetStaticFilesystem() fs.FS
	GetAppFilesystem() fs.FS
Can we replace those two with this one? Or do we need this new one at all? 3 is a crowd :)

Reply:
Those two are *super* finicky and I've spent hours trying and failing to clean them up (something always breaks).

Reply:
For posterity - or to save whoever looks into the next some time: the two existing filesystems are actually performing 3 different functions:
- rendering and serving the "index" file (i.e. the html file that loads the app) at various paths (any path that the app is served from, i.e. `/`, `/trends/`, `/tests/`, `/settings/`, ...)
- serving static files (mostly icons and images) from the /static directory at the root
- serving generated "/app" files i.e. app.js, all of the app bundle chunks, style.css, etc

Reply:
This is awesome, thanks for testing this so thoroughly!